apps/api/pages/api/destination-calendars/[id]/_patch.ts

```diff
@@ -57,13 +60,91 @@ import { schemaQueryIdParseInt } from "~/lib/validations/shared/queryIdTransform
  *        description: Destination calendar not found
  */
 export async function patchHandler(req: NextApiRequest) {
-  const { prisma, query, body } = req;
+  const { userId, isAdmin, prisma, query, body } = req;
   const { id } = schemaQueryIdParseInt.parse(query);
   const parsedBody = schemaDestinationCalendarEditBodyParams.parse(body);
+  let credentialId: number | undefined = undefined;
+  const assignedUserId = isAdmin ? parsedBody.userId || userId : userId;
+
+  let queryConditions = { userId: assignedUserId };
+
+  // when linked with eventTypeId, we need to fetch the userId from the eventTypeId
+  if (parsedBody.eventTypeId) {
+    const eventType = await prisma.eventType.findFirst({
+      where: { id: parsedBody.eventTypeId, userId: assignedUserId },
+    });
+    if (!eventType)
+      throw new HttpError({
+        statusCode: 400,
+        message: "Bad request, eventTypeId invalid",
+      });
+    parsedBody.userId = null;
+    queryConditions = { eventTypeId: parsedBody.eventTypeId };
+  }
+
+  const destinationCalendarObject = await prisma.destinationCalendar.findFirst({
+    where: { id, ...queryConditions },
+    select: { credentialId: true },
+  });
+
+  if (!destinationCalendarObject)
+    throw new HttpError({
+      statusCode: 404,
+      message: `Destination calendar with ID ${id} not found`,
+    });
+
+  if (!destinationCalendarObject.credentialId)
+    throw new HttpError({
+      statusCode: 400,
+      message: `Bad request, credential id invalid`,
+    });
+
+  if (parsedBody.integration && !parsedBody.externalId) {
+    throw new HttpError({ statusCode: 400, message: "External Id is required with integration value" });
+  }
+  if (!parsedBody.integration && parsedBody.externalId) {
+    throw new HttpError({ statusCode: 400, message: "Integration value is required with external ID" });
+  }
+
+  if (parsedBody.integration && parsedBody.externalId) {
+    const userCredentials = await prisma.credential.findMany({
+      where: {
+        type: parsedBody.integration,
+        userId: assignedUserId,
+      },
+      select: credentialForCalendarServiceSelect,
+    });
+
+    if (!userCredentials)
+      throw new HttpError({
+        statusCode: 400,
+        message: "Bad request, credential id invalid",
+      });
+
+    const calendarCredentials = getCalendarCredentials(userCredentials);
+
+    const { connectedCalendars } = await getConnectedCalendars(
+      calendarCredentials,
+      [],
+      parsedBody.externalId
+    );
```

Comment:
If they provide both integration and externalId, it means they wish to update the destinationCalendar credentials, which means they should match up with credential data and should return connected calendars